xbmc/interfaces/legacy/Dialog.cpp

```diff
@@ -238,19 +239,30 @@ namespace XBMCAddon
       std::string value;
       std::string mask = maskparam;
       VECSOURCES *shares = CMediaSourceSettings::GetInstance().GetSources(s_shares);
-      if (!shares) 
-        throw WindowException("Error: GetSources given %s is NULL.",s_shares.c_str());
+
+      VECSOURCES localShares;
+      if (!shares)
+      {
+        g_mediaManager.GetLocalDrives(localShares);
+        if (!s_shares.c_str() || strcmpi(s_shares.c_str(), "local") != 0)
```

Comment:
also strcmpi is a msft-ism, strcasecmp is the posix construct (kodi has defines so it works fine of course so mostly a FYI).